### ISSUE TITLE
linter: false positive with no-standalone-expect

### ISSUE THREAD

**Issue by camc314** [oxc-project/oxc#14888]
> Hi, I'm still seeing this error, my code isn't much different to be honest:
> 
> ```tsx
>   it('should do something', async () => {
>     render(<ExampleComponent />);
> 
>     await waitFor(() => {
>       expect(screen.getByText('Option 2')).toBeInTheDocument();
>     });
>   });
> ```
> 
> I thought it might be because of the async function, but removing that doesn't change anything. 

 _Originally posted by @dan-proudfoot-bots in [#11934](https://github.com/oxc-project/oxc/issues/11934#issuecomment-3426228488)_

**Comment by connorshea** [comment]
I also saw this problem with this jest rule in our codebase, specifically with waitFor(). I'm not sure if the jest eslint plugin suffers from the same problem, though.

**Comment by camc314** [comment]
fixed by #14902
